snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/converter/SnomedConceptConverter.java

```diff
@@ -307,12 +307,24 @@ private void expandDescendants(List<ISnomedConcept> results, final Set<String> c
 				}
 				
 				final int offset = getOffset(expandOptions);
-				final Collection<String> componentIds = newHashSet(descendantsByAncestor.values());
+				final Set<String> componentIds = newHashSet(descendantsByAncestor.values());
 				
 				if (limit > 0 && !componentIds.isEmpty()) {
+					final Set<String> alreadySeenDescendantIds = newHashSet(Sets.intersection(componentIds, conceptIds));
 					// remove any already known concept definition
 					componentIds.removeAll(conceptIds);
```

Comment:
Is it worse from a performance perspective to remove this line instead, and allow the search request to re-fetch and populate the expand fields? It would be easier to read.